File: logic/universal/v1/tools/bridge.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <curl/curl.h>

void query_chatgpt(const char* prompt) {
    CURL *curl = curl_easy_init();
    if(curl) {
        struct curl_slist *headers = NULL;
        char auth_header[256];
        snprintf(auth_header, sizeof(auth_header), "Authorization: Bearer %s", getenv("OPENAI_API_KEY"));
        
        headers = curl_slist_append(headers, "Content-Type: application/json");
        headers = curl_slist_append(headers, auth_header);

        char payload[1024];
        snprintf(payload, sizeof(payload), "{\"model\": \"gpt-4o\", \"messages\": [{\"role\": \"user\", \"content\": \"%s\"}]}", prompt);

        curl_easy_setopt(curl, CURLOPT_URL, "https://openai.com");
        curl_easy_setopt(curl, CURLOPT_HTTPHEADER, headers);
        curl_easy_setopt(curl, CURLOPT_POSTFIELDS, payload);

        curl_easy_perform(curl);
        curl_easy_cleanup(curl);
        curl_slist_free_all(headers);
    }
}

File: logic/universal/v1/tools/bridge.php
<?php
/**
 * AVIS Core v1 - Unified AI Bridge
 */

class AIBridge {
    private $openai_key;
    private $gemini_key;

    public function __construct() {
        $this->openai_key = getenv('OPENAI_API_KEY');
        $this->gemini_key = getenv('GEMINI_API_KEY');
    }

    public function queryChatGPT($prompt, $model = "gpt-4o") {
        $url = "https://openai.com";
        $data = [
            "model" => $model,
            "messages" => [["role" => "user", "content" => $prompt]]
        ];
        
        return $this->executeCurl($url, $data, [
            "Authorization: Bearer " . $this->openai_key,
            "Content-Type: application/json"
        ]);
    }

    public function queryGemini($prompt, $model = "gemini-2.5-flash") {
        $url = "https://googleapis.com" . $model . ":generateContent?key=" . $this->gemini_key;
        $data = [
            "contents" => [["parts" => [["text" => $prompt]]]]
        ];

        return $this->executeCurl($url, $data, ["Content-Type: application/json"]);
    }

    private function executeCurl($url, $data, $headers) {
        $ch = curl_init($url);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
        curl_setopt($ch, CURLOPT_POST, true);
        curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
        $response = curl_exec($ch);
        curl_close($ch);
        return json_decode($response, true);
    }
}
